raiden/transfer/channel.py

```diff
@@ -1351,9 +1352,14 @@ def set_settled(channel_state: NettingChannelState, block_number: BlockNumber) -
         channel_state.settle_transaction.result = TransactionExecutionStatus.SUCCESS
 
 
-def update_contract_balance(end_state: NettingChannelEndState, contract_balance: Balance) -> None:
-    if contract_balance > end_state.contract_balance:
-        end_state.contract_balance = contract_balance
+def update_contract_balance(end_state: NettingChannelEndState, claim: Claim = None) -> None:
+    if claim is None:
+        from raiden.claim import EMPTY_CLAIM
+
+        claim = EMPTY_CLAIM
+
+    if claim.total_amount > end_state.claim.total_amount:
```

Comment:
here we should probably check for a valid signature

Reply:
Not that I disagree, but that might be quite slow as long as we still process all claim on every node.

Reply:
should we do it here or rather before we even create the state change?

Reply:
Doing it with the state change (i.e. `RaidenService.process_claims`) should be fine, I was just triggered by seeing `Claim.amount` amounts being changed in a test

Reply:
i.e. here: https://github.com/raiden-network/raiden/pull/6389/files#diff-f7d7810957ef3a6c3153128196457a74R234

Reply:
amounts of a copy is being changed, isnt it?